File: src/main/java/club/xiaojiawei/custom/LogRunnable.java
package club.xiaojiawei.custom;

import lombok.extern.slf4j.Slf4j;

/**
 * @author 肖嘉威
 * @date 2023/7/8 16:19
 */
@Slf4j
public class LogRunnable implements Runnable{

    private final Runnable task;

    public LogRunnable(Runnable task) {
        this.task = task;
    }

    @Override
    public void run() {
        try {
            task.run();
        }catch (Exception e){
            log.error("Runnable发生错误" , e);
        }
    }
}

File: Hearthstone-Script-Base/src/main/kotlin/club/xiaojiawei/bean/LogRunnable.kt
package club.xiaojiawei.bean

import club.xiaojiawei.config.log

/**
 * @author 肖嘉威
 * @date 2024/9/8 17:02
 */
class LogRunnable(private var task: Runnable?) : Runnable {

    override fun run() {
        try {
            task?.run()
        } catch (e: Exception) {
            if (e is InterruptedException) {
                log.debug(e) { "Runnable发生错误" }
            } else {
                log.error(e) { "Runnable发生错误" }
            }
        }
    }

}